sparkpost/suppression_list.py

```diff
@@ -0,0 +1,82 @@
+import json
+
+from .base import Resource
+
+
+class SuppressionList(Resource):
+    """
+    SuppressionList class used to search, get and modify suppression status.
+    For detailed request and response formats, see the `Suppresion List API
+    documentation<https://www.sparkpost.com/api#/reference/suppression-list>`_.
+    """
+
+    key = 'suppression-list'
+
+    def search(self, **kwargs):
+        """
+        Search for an entry based on the supplied parameters
+
+        :param datetime To: DateTime to end searching
+        :param datetime From: DateTime to start searching
+        :param list Types: Types of entries to include in the search
+        :param int Limit: Maximum number of results to return
+
+        :returns: a ``list`` of entries
+        :raises: :exc:`SparkPostAPIException` if API call fails
+        """
+        params = dict([(i.lower(), kwargs[i]) for i in ["To", "From", "Types",
+                      "Limit"] if i in kwargs])
+        results = self.request('GET', self.uri, params=params)
+        return results
+
+    def check_status(self, email):
+        """
+        Retrieve the suppression status for a specific recipient by email
+
+        :param str email: Email of the recipient whose status you want to
+            check_status
+
+        :returns: a list entry
+        :raises: :exc:`SparkPostAPIException` if API call fails
+        """
+        uri = "%s/%s" % (self.uri, email)
+        results = self.request('GET', uri)
+        return results
+
+    def remove_status(self, email):
+        """
+        Delete the suppression status for a specific recipient by email
+
+        :param str email: Email of the recipient whose status you want to
+            remove
+
+        :returns: TODO
```

Comment:
We can't really do that because of how the base Resource request method works. It doesn't return `status_code`.